experimental/dds/tree2/src/feature-libraries/simple-tree/types.ts

```diff
@@ -23,40 +23,47 @@ import { AssignableFieldKinds } from "../flex-tree";
 
 /**
  * An non-{@link LeafNodeSchema|leaf} SharedTree node. Includes objects, lists, and maps.
+ *
+ * @privateRemarks
+ * This is a union of all possible tree node types.
+ * Since the tree node types are not covariant over their schema, the fact that this works is non-trivial.
+ * TODO: Type tests to ensure that various tree node types are actually assignable to this.
+ *
+ * Using default parameters, this could be combined with TypedNode.
```

Comment:
I was wondering if this union would cause problems. What do you mean, by "could be combined with TypedNode"? Should we do that?

Reply:
TypeNode<TreeNodeSchema>should give this type (it currently doesn't), so if it had a default parameter of TreeNodeSchema, there is no reason to have two separate types.

Reply:
Well, it doesn't give that type because `TypeNode` can be leaf types.